### PR TITLE
[Python][Dev] Switch to using `pyproject.toml` when building and running CI.

### PR DESCRIPTION
**Pull request by Tishj**
This PR should fix Python build issues

### CODE REVIEW

File: .github/workflows/Python.yml [hunk 1]
```diff
@@ -82,7 +82,7 @@ jobs:
       run: |
         export DISTUTILS_C_COMPILER_LAUNCHER=ccache
         # TODO: Use ccache inside container, see https://github.com/pypa/cibuildwheel/issues/1030
-        cibuildwheel --output-dir wheelhouse --config-file cibw.toml duckdb_tarball
+        cibuildwheel --output-dir wheelhouse --config-file pyproject.toml duckdb_tarball
```
Comment (Mause) [inline .github/workflows/Python.yml:85]: Bit of a nitpick, but this is the default value for config-file

**Review by Mause — APPROVED** [review]
Looks good! Just one minor nitpick

File: .github/workflows/Python.yml (hunk 1 above)
Comment (Tishj) [reply .github/workflows/Python.yml:85]: Ah alright, if there are other issues or the PR requires touching in another way I'll address it, if not I'll leave it as is 👍

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**